game-app/game-core/src/main/java/games/strategy/triplea/ai/pro/ProCombatMoveAi.java

```diff
@@ -1197,6 +1156,91 @@ && canAirSafelyLandAfterAttack(unit, t)) {
     }
   }
 
+  private void determineTerritoriesThatCanBeBombed(
+      final Map<Territory, ProTerritory> attackMap,
+      final Map<Unit, Set<Territory>> sortedUnitAttackOptions,
+      final Set<Unit> alreadyAttackedWithUnits) {
+    final boolean raidsMayBePrecededByAirBattles =
+        Properties.getRaidsMayBePreceededByAirBattles(data.getProperties());
+    territoryManager
+        .getAttackOptions()
+        .getBomberMoveMap()
+        .entrySet()
+        .forEach(
+            bomberEntry -> {
+              final Unit bomber = bomberEntry.getKey();
+              if (alreadyAttackedWithUnits.contains(bomber)) {
+                return; // already attacked bombers cannot move
+              }
+              Set<Territory> bomberTargetTerritories = bomberEntry.getValue();
+              if (raidsMayBePrecededByAirBattles) {
+                bomberTargetTerritories =
+                    new HashSet(
+                        CollectionUtils.getMatches(
+                            bomberTargetTerritories,
+                            terr ->
+                                AirBattle.territoryCouldPossiblyHaveAirBattleDefenders(
+                                    terr, player, data, true)));
```

Comment:
I don't think we can check `RAIDS_MAY_BE_PRECEEDED_BY_AIR_BATTLES` at the beginning of the method, because we checked !(1 && 2), which is the same as (!1 || !2), so in particular, if 2 is false (there are no potential air units for the battle), we would execute the code.

Reply:
So we want to avoid air battles: If the property allows preceding air battels, only territories without the potential to have one should be kept, else all territories are relevant.
I made a change and added a comment for this.
Note that I use a not now: `!AirBattle.territoryCouldPossiblyHaveAirBattleDefenders`